contracts/ethregistrar/ETHRegistrarController.sol

```diff
@@ -1,178 +1,299 @@
         commitments[commitment] = block.timestamp;
     }
 
-    function register(string calldata name, address owner, uint duration, bytes32 secret) external payable {
-      registerWithConfig(name, owner, duration, secret, address(0), address(0));
-    }
-
-    function registerWithConfig(string memory name, address owner, uint duration, bytes32 secret, address resolver, address addr) public payable {
-        bytes32 commitment = makeCommitmentWithConfig(name, owner, secret, resolver, addr);
-        uint cost = _consumeCommitment(name, duration, commitment);
-
+    function register(
+        string calldata name,
+        address owner,
+        bytes32 secret,
+        address resolver,
+        bytes[] calldata data,
+        bool reverseRecord,
+        uint96 fuses
+    ) public payable override {
         bytes32 label = keccak256(bytes(name));
-        uint256 tokenId = uint256(label);
-
-        uint expires;
-        if(resolver != address(0)) {
-            // Set this contract as the (temporary) owner, giving it
-            // permission to set up the resolver.
-            expires = base.register(tokenId, address(this), duration);
-
-            // The nodehash of this label
-            bytes32 nodehash = keccak256(abi.encodePacked(base.baseNode(), label));
-
-            // Set the resolver
-            base.ens().setResolver(nodehash, resolver);
-
-            // Configure the resolver
-            if (addr != address(0)) {
-                Resolver(resolver).setAddr(nodehash, addr);
-            }
-
-            // Now transfer full ownership to the expeceted owner
-            base.reclaim(tokenId, owner);
-            base.transferFrom(address(this), owner, tokenId);
-        } else {
-            require(addr == address(0));
-            expires = base.register(tokenId, owner, duration);
+        (uint256 duration, uint256 premiumCost) = prices.duration(
+            name,
+            base.nameExpires(uint256(label)),
+            msg.value
+        );
+        _consumeCommitment(
+            name,
+            duration,
+            makeCommitment(
+                name,
+                owner,
+                secret,
+                resolver,
+                data,
+                reverseRecord,
+                fuses
+            )
+        );
+
+        uint256 expires = nameWrapper.registerAndWrapETH2LD(
+            name,
+            owner,
+            duration,
+            resolver,
+            fuses
+        );
+
+        if (data.length > 0) {
+            require(
+                resolver != address(0),
+                "ETHRegistrarController: resolver is required when data is supplied"
+            );
+            _setRecords(resolver, label, data);
         }
 
-        emit NameRegistered(name, label, owner, cost, expires);
-
-        // Refund any extra payment
-        if(msg.value > cost) {
-            payable(msg.sender).transfer(msg.value - cost);
```

Comment:
why do we no longer do this at registration? We set 10% higher to accommodate for price oracle change

Reply:
See below:

Reply:
What am I seeing below?